cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -157,14 +161,68 @@ public Void call() throws Exception {
         authorizerInstantiator.get().revoke(namespace);
       }
       throw new NamespaceCannotBeCreatedException(namespace.toId(), e);
     }
+    nsStore.create(metadata);
+  }
+
+  private void validateCustomMapping(NamespaceMeta metadata) throws Exception {
+    for (NamespaceMeta existingNamespaceMeta : list()) {
+      NamespaceConfig existingConfig = existingNamespaceMeta.getConfig();
+      // if hbase namespace is provided validate no other existing namespace is mapped to it
+      if (!Strings.isNullOrEmpty(metadata.getConfig().getHbaseNamespace()) &&
+        metadata.getConfig().getHbaseNamespace().equals(existingConfig.getHbaseNamespace())) {
+        throw new NamespaceAlreadyExistsException(existingNamespaceMeta.getNamespaceId().toId(),
+                                                  String.format("A namespace '%s' already exists with the given " +
+                                                                  "namespace mapping for hbase namespace '%s'",
+                                                                existingNamespaceMeta.getName(),
+                                                                existingConfig.getHbaseNamespace()));
+      }
+      // if hive database is provided validate no other existing namespace is mapped to it
+      if (!Strings.isNullOrEmpty(metadata.getConfig().getHiveDatabase()) &&
+        metadata.getConfig().getHiveDatabase().equals(existingConfig.getHiveDatabase())) {
+        throw new NamespaceAlreadyExistsException(existingNamespaceMeta.getNamespaceId().toId(),
+                                                  String.format("A namespace '%s' already exists with the given " +
+                                                                  "namespace mapping for hive database '%s'",
+                                                                existingNamespaceMeta.getName(),
+                                                                existingConfig.getHiveDatabase()));
+      }
+      if (!Strings.isNullOrEmpty(metadata.getConfig().getRootDirectory())) {
+        // check that the given root directory path is an absolute path
+        hasAbsolutePath(metadata);
+        if (isSubdir(existingConfig.getRootDirectory(), metadata.getConfig().getRootDirectory())) {
+          throw new NamespaceAlreadyExistsException(existingNamespaceMeta.getNamespaceId().toId(),
+                                                    String.format("A namespace '%s' already exists with the given " +
+                                                                    "namespace mapping for root directory '%s'",
+                                                                  existingNamespaceMeta.getName(),
+                                                                  existingConfig.getRootDirectory()));
+        }
+      }
+    }
+  }
 
+  private boolean isSubdir(@Nullable String existingDir, String newDir) {
```

Comment:
Yes because all the content of cdap namespace will be in the child dir. If We don't allow that then all custom mapping will be /firstlevel and nothing else will be possible.